### PR TITLE
Bitcoin Cash support

### PR DESCRIPTION
**Pull request by isocolsky**

### CODE REVIEW

**Comment by matiu** [comment]
ACK

**Comment by nicsu82** [comment]
thx..

**Comment by matiu** [comment]
Great work!

some comments:
1) Storage.prototype.storeAddress = function(address, cb) {
show receive coin or use address.coin in the selection

2) Storage.prototype.storeAddressAndWallet = function(wallet, addresses, cb) {
    self.db.collection(collections.ADDRESSES).findOne({
should use coin

3) Storage.prototype.fetchAddress = function(address, cb) {

should use coin?

4) We should review Storage.prototype._createIndexes = function() {
maybe due to 2) we need to add  address,coin index?

**Comment by isocolsky** [comment]
Thank you for the review. All concerns in your comment were addressed. There is no longer a `fetchAddress` method. It was replaced with `fetchAddressByWalletId` & `fetchAddressByCoin` to avoid confusion. Also, `storeAddressAndWallet` no longer checks for unique addresses among wallets.

**Comment by martindale** [comment]
How was this pull request accepted without community approval?  :confused: 

As a shareholder, this is both confusing and disappointing.  Instantaneous payments, not increased block size, will create a consumer payment experience worth adoption.

File: config.js [hunk 1]
```diff
@@ -38,16 +38,28 @@ var config = {
...
-    testnet: {
-      provider: 'insight',
-      url: 'https://test-insight.bitpay.com:443',
-      // url: 'http://localhost:3001',
-      // Multiple servers (in priority order)
-      // url: ['http://a.b.c', 'https://test-insight.bitpay.com:443'],
+    bch: {
+      livenet: {
+        provider: 'insight',
+        url: 'https://cashexplorer.bitcoin.com',
```
Comment (martindale) [inline config.js]: This is an external service; strongly recommend hosting on @BitPay's servers, but if not, please notify users of the risks before deploying this code to production.

**Comment by dabura667** [comment]
I'm guessing the decision is "if 2x flops Bitpay is switching to BCash"

Good luck with that.

**Pull request merged**